lnwallet/wallet.go

```diff
@@ -752,15 +759,34 @@ func (l *LightningWallet) handleFundingReserveRequest(req *InitFundingReserveMsg
 	// funder in the attached request to provision the inputs/outputs
 	// that'll ultimately be used to construct the funding transaction.
 	if !ok {
+		var err error
+		var numAnchorChans int
+
+		// Get the number of anchor channels to determine if there is a reserved
+		// value that must be respected when funding up to the maximum amount.
+		if req.FundUpToMaxAmt > 0 {
+			numAnchorChans, err = l.currentNumAnchorChans()
+			if err != nil {
+				req.err <- err
+				req.resp <- nil
+				return
+			}
+
+			if hasAnchors {
+				numAnchorChans++
+			}
+		}
+
 		// Coin selection is done on the basis of sat/kw, so we'll use
 		// the fee rate passed in to perform coin selection.
-		var err error
 		fundingReq := &chanfunding.Request{
-			RemoteAmt:    req.RemoteFundingAmt,
-			LocalAmt:     req.LocalFundingAmt,
-			MinConfs:     req.MinConfs,
-			SubtractFees: req.SubtractFees,
-			FeeRate:      req.FundingFeePerKw,
+			RemoteAmt:      req.RemoteFundingAmt,
+			LocalAmt:       req.LocalFundingAmt,
+			FundUpToMaxAmt: req.FundUpToMaxAmt,
+			ReservedAmt:    ReservedValue(numAnchorChans),
```

Comment:
This will lead to inconsistent behavior.
For private channels (i.e mobile wallets), we do not right now reserve any sats for anchor outputs.
If you specify the amount when opening a private channel, you'll be able to use all funds (no sats will be reserved).
Related commit: https://github.com/lightningnetwork/lnd/commit/8e1087d1cdcbb21b495cffa74ac0afc6c2ffa533

Also related topic: https://github.com/lightningnetwork/lnd/issues/6574

Reply:
Addressed and described here with comments from the related commit: https://github.com/lightningnetwork/lnd/pull/6903/commits/60ddaa719a0e36f104304ffa9adb901764c9e314#diff-3ebebe355b386e69fdd9742a8d6ed60cb3b2377a9c10a3d3b831650e47c7d258R779